### ISSUE TITLE
使用提供的config训练2000轮达不到benchmark上的指标

### ISSUE THREAD

**Issue by Secilia-Cxy** [chengtan9907/OpenSTL#99]
作者您好，我在研究SimVP相关的工作，尝试复现MMNIST数据集上的MetaFormer结果。我在单gpu上使用了代码库中提供的config，发现不能达到benchmark中的结果，且结果相差还比较大（如benchmark中SimVP-gSTA的MSE可以达到16以下，但是在实验中MSE为20.9）。想问一下在训练中是否有参数需要修改还是可能是别的问题导致的这个原因。谢谢！
![image](https://github.com/chengtan9907/OpenSTL/assets/67688986/743cfd5c-fcf7-4fdd-a82f-8cd2459bf3a0)
![image](https://github.com/chengtan9907/OpenSTL/assets/67688986/36bfe2bf-a47f-4374-b1bb-07a124c1690a)

**Comment by chengtan9907** [comment]
能否把log文件中环境、config部分的信息贴出来看一下呢？

**Comment by Secilia-Cxy** [comment]
请问是这部分吗?
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
Global seed set to 42
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/1
----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All distributed processes registered. Starting with 1 processes
----------------------------------------------------------------------------------------------------
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]
Environment info:
------------------------------------------------------------
sys.platform: linux
Python: 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0]
CUDA available: True
CUDA_HOME: /usr/local/cuda
NVCC: Build cuda_12.1.r12.1/compiler.32688072_0
GPU 0,1,2,3,4,5,6,7: NVIDIA H100 80GB HBM3
GCC: gcc (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0
PyTorch: 2.1.0a0+b5021ba
PyTorch compiling details: PyTorch built with:
  - GCC 11.3
  - C++ Version: 201703
  - Intel(R) oneAPI Math Kernel Library Version 2021.1-Product Build 20201104 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.7.3 (Git Hash N/A)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 12.1
  - NVCC architecture flags: -gencode;arch=compute_52,code=sm_52;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_90,code=sm_90;-gencode;arch=compute_90,code=compute_90
  - CuDNN 8.9.3
  - Magma 2.6.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=12.1, CUDNN_VERSION=8.9.3, CXX_COMPILER=/usr/bin/c++, CXX_FLAGS=-fno-gnu-unique -D_GLIBCXX_USE_CXX11_ABI=1 -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=range-loop-construct -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=old-style-cast -Wno-invalid-partial-specialization -Wno-unused-private-field -Wno-aligned-allocation-unavailable -Wno-missing-braces -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -f (line truncated to 1000 characters)
TorchVision: 0.16.0a0
OpenCV: 4.7.0
openstl: 1.0.0
------------------------------------------------------------
device: 	cuda	
dist: 	False	
res_dir: 	work_dirs	
ex_name: 	mmnist_simvp_gSTA	
fp16: 	False	
torchscript: 	False	
seed: 	42	
fps: 	False	
test: 	False	
deterministic: 	False	
batch_size: 	16	
val_batch_size: 	16	
num_workers: 	4	
data_root: 	./data	
dataname: 	mmnist	
pre_seq_length: 	10	
aft_seq_length: 	10	
total_length: 	20	
use_augment: 	False	
use_prefetcher: 	False	
drop_last: 	False	
method: 	simvp	
config_file: 	configs/mmnist/simvp/SimVP_gSTA.py	
model_type: 	gSTA	
drop: 	0.0	
drop_path: 	0	
overwrite: 	False	
epoch: 	2000	
log_step: 	1	
opt: 	adam	
opt_eps: 	None	
opt_betas: 	None	
momentum: 	0.9	
weight_decay: 	0.0	
clip_grad: 	None	
clip_mode: 	norm	
no_display_method_info: 	False	
sched: 	onecycle	
lr: 	0.001	
lr_k_decay: 	1.0	
warmup_lr: 	1e-05	
min_lr: 	1e-06	
final_div_factor: 	10000.0	
warmup_epoch: 	0	
decay_epoch: 	100	
decay_rate: 	0.1	
filter_bias_and_bn: 	False	
gpus: 	[0]	
metric_for_bestckpt: 	val_loss	
spatio_kernel_enc: 	3	
spatio_kernel_dec: 	3	
hid_S: 	64	
hid_T: 	512	
N_T: 	8	
N_S: 	4	
in_shape: 	[10, 1, 64, 64]	
data_name: 	mnist	
metrics: 	['mse', 'mae', 'ssim', 'psnr']

**Comment by chengtan9907** [comment]
log文件中模型部分的可以再补充一下嘛？

**Comment by Secilia-Cxy** [comment]
非常抱歉刚刚贴错了！以下是最开始跑的200epoch的实验，也是同样的问题。麻烦您看一下这个，test MSE是39.23，但是表格中是26.69。
![image](https://github.com/chengtan9907/OpenSTL/assets/67688986/983a75a3-23ad-40c9-b8ef-4464ddc6aa9e)
[output.log](https://github.com/chengtan9907/OpenSTL/files/14066030/output.log)

**Comment by chengtan9907** [comment]
环境和模型看上去没什么问题，但您给的output.log文件看上去很奇怪？似乎只有Epoch 0的信息， train_loss_step停留在0.0506？

**Comment by Secilia-Cxy** [comment]
这个我也不知道是什么原因，但是wandb上的log是正常的，我之前check了一下您提供的log文件，发现相比之下我跑的val_loss也下降的有点慢，是不是我学习率设置的不对？
![image](https://github.com/chengtan9907/OpenSTL/assets/67688986/5f727544-c409-4ac8-a625-79b0f1282f09)

**Comment by chengtan9907** [comment]
学习率看上去也没什么问题，您可以把./word_dirs下对应实验的完整log文件贴上来看看

**Comment by Secilia-Cxy** [comment]
刚刚贴的是wandb记录的log文件，这个是您说的目录下的完整的log文件。
[train_20240124_155508.log](https://github.com/chengtan9907/OpenSTL/files/14066354/train_20240124_155508.log)

**Comment by chengtan9907** [comment]
您log文件中的lr似乎有问题，我们用lr scheduler，每轮不应该是固定的0.0000401。我最近跑的前几轮lr变化是这样的：

```
2023-11-27 08:18:18,049 - Epoch 1: Lr: 0.0000426 | Train Loss: 0.0347514 | Vali Loss: 0.0278715
2023-11-27 08:19:27,991 - Epoch 2: Lr: 0.0000459 | Train Loss: 0.0284847 | Vali Loss: 0.0262449
2023-11-27 08:20:39,586 - Epoch 3: Lr: 0.0000505 | Train Loss: 0.0267219 | Vali Loss: 0.0248762
2023-11-27 08:21:52,459 - Epoch 4: Lr: 0.0000564 | Train Loss: 0.0254619 | Vali Loss: 0.0237298
2023-11-27 08:23:08,982 - Epoch 5: Lr: 0.0000635 | Train Loss: 0.0242142 | Vali Loss: 0.0230346
```

**Comment by Secilia-Cxy** [comment]
确实是您说的问题，但是我并没有修改lr scheduler相关的代码，只修改了logger。请问您知道可能是什么操作导致的这个问题吗？

**Comment by chengtan9907** [comment]
我不太确定具体是什么原因，我建议您重新拉取我们最新的代码，最好在确认跑通之后再作改动

**Comment by Secilia-Cxy** [comment]
感谢您的建议！我刚刚重新clone了default分支下的最新版本代码，且重新pip install -e .，运行了一下，还是有一样的问题。
[train_20240126_234817.log](https://github.com/chengtan9907/OpenSTL/files/14066836/train_20240126_234817.log)

**Comment by chengtan9907** [comment]
这有点奇怪，我这是没问题的。我猜测可能是Lightning版本的问题。

**Comment by chengtan9907** [comment]
I will close this issue if there are no more questions. Please feel free to open a new issue when you have new questions.

**Comment by chengtan9907** [comment]
Sorry for the confusion, @Secilia-Cxy. Please refer to the [issue#113](https://github.com/chengtan9907/OpenSTL/issues/113).

**Comment by chengtan9907** [comment]
I have fixed this bug in the commit [a6e0dd3](https://github.com/chengtan9907/OpenSTL/commit/a6e0dd3cc0a3aa8ad940c3f972d05e57ff74e4f0).

**Comment by Secilia-Cxy** [comment]
Thank you! My problem has been solved!

**Comment by WangDongBUAA** [comment]
> 作者您好，我在研究SimVP相关的工作，尝试复现MMNIST数据集上的MetaFormer结果。我在单gpu上使用了代码库中提供的config，发现不能达到benchmark中的结果，且结果相差还比较大（如benchmark中SimVP-gSTA的MSE可以达到16以下，但是在实验中MSE为20.9）。想问一下在训练中是否有参数需要修改还是可能是别的问题导致的这个原因。谢谢！ ![image](https://github.com/chengtan9907/OpenSTL/assets/67688986/743cfd5c-fcf7-4fdd-a82f-8cd2459bf3a0) ![image](https://github.com/chengtan9907/OpenSTL/assets/67688986/36bfe2bf-a47f-4374-b1bb-07a124c1690a)



您好，请问下这个图是怎么得到的呢？关于MSE、SSIM这些指标是怎么得来的？

感谢！
